test/metrics/metrics_test.go

```diff
@@ -0,0 +1,222 @@
+package metrics_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+
+	"github.com/orange-cloudavenue/kube-image-updater/internal/metrics"
+)
+
+func TestMetric_Counter(t *testing.T) {
+	// Test the metrics for the actions Counter
+	list := metrics.Metrics
+
+	// Test the metrics for the actions Counter
+	type testsCounter []struct {
+		name       string
+		nameMetric string
+		data       string
+		value      string
+		c          prometheus.Counter
+		error      bool
+	}
+	testUnit := make(testsCounter, 0)
+
+	// loop over the list of metrics
+	for _, m := range list[metrics.MetricTypeCounter] {
+		// Check if the metric is a metricCounter
+		if m, ok := m.(metrics.MetricCounter); ok {
+			// TODO - Constructs all the expected tests
+			// fill struct test with data
+			testUnit = testsCounter{
+				{
+					name:       "Check Counter " + m.Name,
+					nameMetric: m.Name,
+					data: fmt.Sprintf(`
+# HELP %s %s
+# TYPE %s %s
+`, m.Name, m.Help, m.Name, metrics.MetricTypeCounter),
+					value: " 1\n",
+					c:     *m.Counter,
+					error: false,
+				},
+				{
+					name:       "Check Counter mistake between name and TYPE description " + m.Name,
+					nameMetric: m.Name,
+					data: fmt.Sprintf(`
+# HELP %s %s
+# TYPE %s_mistake_error_in_TYPE %s
+`, m.Name, m.Help, m.Name, metrics.MetricTypeCounter),
+					value: " 1\n",
+					error: true, // Error because the counter name is not the same in the HELP description
+				},
+				{
+					name:       "Check Counter mistake between name and HELP description " + m.Name,
+					nameMetric: m.Name,
+					data: fmt.Sprintf(`
+# HELP %s_mistake_error_in_HELP %s
+# TYPE %s %s
+`, m.Name, m.Help, m.Name, metrics.MetricTypeCounter),
+					value: " 1\n",
+					error: true, // Error because the counter name is not the same in the description
+				},
```

Comment:
```suggestion
				{
					name:       "Check Counter mistake between name and TYPE description " + m.Name,
					nameMetric: m.Name,
					data: fmt.Sprintf(`
# HELP %s %s
# TYPE %s_mistake_error_in_TYPE %s
`, m.Name, m.Help, m.Name, metrics.MetricTypeCounter),
					value: " 1\n",
					c: *m.Counter,
					error: true, // Error because the counter name is not the same in the HELP description
				},
				{
					name:       "Check Counter mistake between name and HELP description " + m.Name,
					nameMetric: m.Name,
					data: fmt.Sprintf(`
# HELP %s_mistake_error_in_HELP %s
# TYPE %s %s
`, m.Name, m.Help, m.Name, metrics.MetricTypeCounter),
					value: " 1\n",
					c: *m.Counter,
					error: true, // Error because the counter name is not the same in the description
				},
```